### PR TITLE
sp-version: Add some more docs.

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: primitives/version/src/lib.rs [hunk 1]
```diff
@@ -15,7 +15,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Version module for the Substrate runtime; Provides a function that returns the runtime version.
+//! Substrate runtime version
+//!
+//! Each runtime that should be executed by a Substrate based node needs to have a runtime version.
+//! The runtime version is defined by [`RuntimeVersion`]. The runtime version is used to
+//! distinguish different runtimes, while the most important field is the
```
Comment (shawntabrizi) [inline primitives/version/src/lib.rs:22]:
```suggestion
//! distinguish different runtimes. The most important field is the
```

File: primitives/version/src/lib.rs [hunk 2]
```diff
@@ -15,7 +15,21 @@
...
+//! [`spec_version`](RuntimeVersion::spec_version). The `spec_version` should be increased in a
+//! runtime when a new runtime build includes breaking changes that would make other runtimes unable
+//! to import blocks build by this runtime or vice-versa the new runtime could not import blocks
```
Comment (shawntabrizi) [inline primitives/version/src/lib.rs:25]:
```suggestion
//! to import blocks built by this runtime, or vice-versa where the new runtime could not import blocks
```

File: primitives/version/src/lib.rs [hunk 3]
```diff
@@ -15,7 +15,21 @@
...
+//! build by the old runtime. The runtime version also carries other version information about the
```
Comment (shawntabrizi) [inline primitives/version/src/lib.rs:26]:
```suggestion
//! built by the old runtime. The runtime version also carries other version information about the
```

File: primitives/version/src/lib.rs [hunk 4]
```diff
@@ -15,7 +15,21 @@
...
+//! runtime, see [`RuntimeVersion`] for more information on this.
+//!
+//! Substrate will fetch the runtime version from a `wasm` blob by first checking the
+//! `runtime_version` link section or calling the `Core::version` runtime api. The link section can
+//! be generated in the runtime using the [`runtime_version`] attribute. The `Core` runtime api also
+//! has to be implemented for the runtime using `impl_runtime_apis!`.
```
Comment (shawntabrizi) [inline primitives/version/src/lib.rs:32]:
```suggestion
//! needs to be implemented for the runtime using `impl_runtime_apis!`.
```

**Review by shawntabrizi — APPROVED** [review]

**Review by NukeManDan — APPROVED** [review]
Awesome!
It might be a but over verbose, but far better than under verbose. Good addition for sure!

**Pull request merged**